### PR TITLE
[Misc] Suppress dgl warning in unit tests.

### PR DESCRIPTION
**Pull request by frozenbugs**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/base.py [hunk 1]
```diff
@@ -50,6 +51,9 @@ def dgl_warning_format(message, category, filename, lineno, line=None):
 
 def dgl_warning(message, category=DGLWarning, stacklevel=2):
     """DGL warning wrapper that defaults to ``DGLWarning`` instead of ``UserWarning`` category."""
+    # Skip warning in unit test.
+    if "unittest" in sys.modules.keys():
```
Comment (Rhett-Ying) [inline python/dgl/base.py:55]: skip all warnings is too mighty...

**Review by Rhett-Ying — APPROVED** [review]

File: python/dgl/base.py (hunk 1 above)
Comment (frozenbugs) [reply python/dgl/base.py:55]: The problem is "did we ever care about warnings in unit test", I can move this to the item_sampler.py related to https://github.com/dmlc/dgl/pull/7047, and we can see whether the warning in unit test is helpful or just more trouble.

Comment (Rhett-Ying) [reply python/dgl/base.py:55]: I think we should care about warnings even in unit test in default as it reminds us we may call APIs in not-recommended way. For those we really don't care about it, we'd better capture warning explicitly right in the testcase which indicates: we know warning is thrown and we don't care about it. As the `timestamp` warning in #7047, I am wondering should we add it into canonical attribute list to suppress the warning.

Comment (mfbalin) [reply python/dgl/base.py:55]: I caught some bugs thanks to the warnings in the tests.

**Pull request closed without merging**